### ISSUE TITLE
FR: Add loss weighting for imbalanced Dataset

### ISSUE THREAD

**Issue by stroblme** [cirKITers/Split-Optimization#13]
**Is your feature request related to a problem? Please describe.**
MNIST has a different number of samples for each class.
Imbalanced data can be a hamper the test performance.

**Describe the solution you'd like**
Introduce a weighting of the loss based on the number of samples available for each class.
This is available as a parameter in [PyTorch's CE Loss Method](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss)

**Describe alternatives you've considered**
Oversampling: Introduced duplicate samples; slows down training; pot. overfitting
Undersampling: Removing samples can cause the model to miss important features

**Additional context**
N/A
